src/art.rs

```diff
@@ -1041,9 +1020,20 @@ impl<P: KeyTrait, V: Clone> Tree<P, V> {
         Tree {
             root: None,
             size: 0,
+            version: 0,
         }
     }
 
+    fn update_version(&mut self, version: u64) {
```

Comment:
TODO: Add comment for this changed version behaviour which just moves forward in time but doesn't get updated upon any removals. This may not be useful for external users, but internally in surrealkv, we use this version to for snapshot reads. And they always have to move forward in time, as a deletion shouldn't roll back a previous snapshot.